omod/src/main/java/org/openmrs/web/controller/PortletController.java

```diff
@@ -202,36 +208,16 @@ public ModelAndView handleRequest(HttpServletRequest request, HttpServletRespons
 					}
 					
 					if (Context.hasPrivilege(PrivilegeConstants.GET_OBS)) {
-						// Get pagination parameters
-						Integer page = getPageParameter(request);
 						Integer pageSize = getPageSizeParameter(request, as);
 						
 						Person person = (Person) p;
 						List<Person> persons = Collections.singletonList(person);
 						
-						// Setup parameters for database-level pagination
-						List<Encounter> encounters = null;
-						List<Concept> questions = null;
-						List<Concept> answers = null;
-						List<PERSON_TYPE> personTypes = null;
-						List<Location> locations = null;
-						List<String> sort = Collections.singletonList("obsDatetime desc");
-						Integer obsGroupId = null;
-						Date fromDate = null;
-						Date toDate = null;
-						boolean includeVoided = false;
-						
-						// Get observations for the current page
-						List<Obs> paginatedObs = Context.getObsService().getObservations(persons, encounters, questions,
-						    answers, personTypes, locations, sort, null, obsGroupId, fromDate, toDate, includeVoided);
-						
-						// Get total count for pagination
-						Integer totalCount = Context.getObsService().getObservationCount(persons, encounters, questions,
-						    answers, personTypes, locations, obsGroupId, fromDate, toDate, includeVoided);
+						List<Obs> paginatedObs = Context.getObsService().getObservations(persons, null, null, null, null,
+						    null, Collections.singletonList("obsDatetime desc"), pageSize,
```

Comment:
Can we apply the same concept here?

Reply:
So I am using the using proper ObsService method signature for observations and using mostRecentN for pagination with the pageSize parameter, using same pageSize from global property or default which is set to 50 thus showing most recent observations up to pageSize limit